### PR TITLE
Adding changes to implement version display in iProduct USB desciptor…

### PR DESCRIPTION
**Pull request by scott-berman-brt**
… field

### CODE REVIEW

File: build_all.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
Comment (brian-brt) [inline build_all.sh]: This is unrelated, please move to a separate PR.

The rest of this looks good to submit upstream. This script does not though.

File: include/config.h [hunk 2]
```diff
@@ -38,11 +38,15 @@ THE SOFTWARE.
 #define BOARD_cantact     2
 #define BOARD_canable     3
 #define BOARD_usb2can     4
-#define BOARD_canalyze     5
+#define BOARD_canalyze    5
 #define BOARD_cannette    6
 
 #if BOARD == BOARD_candleLight
+#ifdef VERSION_STR
+	#define USBD_PRODUCT_STRING_FS		(uint8_t*) VERSION_STR
```
Comment (brian-brt) [inline include/config.h]: How about appending instead? Something like this up at the top:
```
#ifdef VERSION_STR
#define PRODUCT_APPEND_STR " " VERSION_STR
#else
#define PRODUCT_APPEND_STR ""
#endif
```

And then for each one it just turns into:
```
#define USBD_PRODUCT_STRING_FS		(uint8_t*) ("candleLight USB to CAN adapter" PRODUCT_APPEND_STR)
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -101,25 +111,37 @@ endfunction()
...
     target_link_libraries(${TGTNAME}_fw PRIVATE STM32_HAL_STM32F042x6 STM32_USB_Device_Library_STM32F042x6)
     make_bin_file(${TGTNAME}_fw)
     dfu_flash(${TGTNAME}_fw)
     show_object_size(${TGTNAME}_fw)
+    append_version(${TGTNAME}_fw ${TGTNAME}-${versionStr})
 endfunction()
 
 function(add_f072_target TGTNAME)
+    execute_process(
+        COMMAND git -C ${CMAKE_CURRENT_BINARY_DIR} log -1 --format=%h
```
Comment (brian-brt) [inline CMakeLists.txt:132]: `git rev-parse HEAD` is shorter if you want.

File: build_all.sh (hunk 1 above)
Comment (scott-berman-brt) [reply build_all.sh]: Done.  I'll move to separate PR

File: include/config.h (hunk 2 above)
Comment (scott-berman-brt) [reply include/config.h]: This is how the version appears after my change.  The name is the same as the filename without the .bin extension.  @brian-brt: Do you think I should append this to the end of the version?

$ ./read_version.sh 
canable-d94ccc6f

File: CMakeLists.txt (hunk 3 above)
Comment (scott-berman-brt) [reply CMakeLists.txt:132]: git rev-parse HEAD generates the full hash.  I was using the short - first 7 characters - for a version.
```
$ git rev-parse HEAD
cec596c4231b8b47441291f89f879119d0decf83
```

File: include/config.h (hunk 2 above)
Comment (brian-brt) [reply include/config.h]: I think it's useful to include the string still, so you can tell what hardware the firmware is built for.

File: include/config.h [hunk 4]
```diff
@@ -38,11 +38,17 @@ THE SOFTWARE.
...
 #define BOARD_canable     3
 #define BOARD_usb2can     4
-#define BOARD_canalyze     5
+#define BOARD_canalyze    5
 #define BOARD_cannette    6
 
+#ifdef VERSION_STR
+    #define PRODUCT_APPEND_STR " " VERSION_STR
+#else
+    #define PRODUCT_APPEND_STR ""
```
Comment (brian-brt) [inline include/config.h]: The `""` doesn't do anything, and it's kind of confusing to have it.

**Review by brian-brt — COMMENTED** [review]
words because github makes me type something

File: include/config.h (hunk 2 above)
Comment (scott-berman-brt) [reply include/config.h]: Here's the USBD_PRODUCT_STRING  
```
$ sudo lsusb -d 1d50:606f -v |grep iProduct
  iProduct                2 canable gs_usb canable-f58769b
```
The first two words are same as original and the third is the version string which matches the filename.

File: include/config.h (hunk 4 above)
Comment (scott-berman-brt) [reply include/config.h]: Removed ""

**Review by brian-brt — APPROVED** [review]
